querymsc.py

```diff
@@ -505,5 +507,124 @@ def query_rain_rate_one_day_1_50(self, session):
                                   History.lon,
                                   History.station_id)
                  )
- 
-        return query
\ No newline at end of file
+  
+        return query
+
+class Gumbel:
+    """This class is meant to do all the fitting
+    of Gumbel distributions to extract design values.
+    
+    Attributes:
+        return_period (int): return period in years
+        min_fit (int): minimum number of years to consider
+            fitting Gumbel.
+    """
+
+    def __init__(self, return_period, min_fit=10):
+
+        self.return_period = return_period
+        self.min_fit = min_fit
+
+    def fit(self, x):
+        """Function to get L-moments to estimate the parameters
+        of a gumbel distribution. This method uses the 
+        right-skewed Gumbel distribution. Method mirrors 
+        Hosking 1990.
+        -----------------------------------
+        Args:
+            x (pandas Series): Series containing the annual
+                grouped extreme values for a given 
+                station over a range of years.
+        Returns:
+            (xi, alpha) (tuple): estimated parameters of gumbel
+                distribution if N_min criteria met
+            NaN (numpy NaN object): if N_min criteria is not met 
+        """
+
+        N = x.shape[0]
+
+        # euler-mascheroni constant
+        euler = 0.5772156649
+
+        if N >= self.min_fit:
+            # create fitted gumbel object to
+            # distribution of extreme values
+            paras = distr.gum.lmom_fit(x)
+
+            # extract L-moments
+            lmoments = distr.gum.lmom(nmom=2, **paras)
+
+            # calculate estimators of gumbel
+            # parameters
+            alpha = lmoments[1]/np.log(2)
+            xi = lmoments[0] - euler*alpha 
+
+            return xi, alpha
+        else:
+            return np.nan, np.nan
```

Comment:
This is an OK way to handle the cases where there are not enough observations, but think about what the cases are that are going to have to handle the results of this computation. Are you now going to have to check for `np.nan` everywhere you use the results? Further, lIttering your code with if/thens all over the place? Is this actually an error condition? Like should the computation stop at this point? Maybe it's appropriate to raise an `Exception` and "fail early, fail often". I'm not sure of the answer, but I'm not sure how useful a `NaN` return value is.

Reply:
This was my intended usage, and it was leveraging pandas. If `df_one_day` contains the grouped stations, then

```
gumbel = Gumbel(10, min_fit=10)
gumbel.fit_transform(df=df_one_day).dropna()
```

Then `gumbel.fit_transform(df=df_one_day).dropna()` gives a new pandas dataframe that contains a "good" fit, or else the row contains NaN for the new design value.